core/templates/pages/exploration-player-page/services/question-player-engine.service.spec.ts

```diff
@@ -743,6 +745,41 @@ describe('Question player engine service ', () => {
       expect(questionPlayerEngineService.getCurrentIndex()).toBe(1);
     });
 
+    it('should create next question if the existing ' +
+      'question is not the last one', () => {
+      let successCallback = jasmine.createSpy('success');
+      let successHandler = jasmine.createSpy('success');
+      let failHandler = jasmine.createSpy('fail');
+      let answer = 'answer';
+      let interactionRulesService: InteractionRulesService;
+      let answerClassificationResult = new AnswerClassificationResult(
+        outcomeObjectFactory
+          .createNew('default', '', '', []), 1, 0, 'default_outcome'
+      );
+      let sampleCard = stateCardObjectFactory.createNewCard(
+        'Card 1', 'Content html', 'Interaction text', null,
+        null, null, 'content_id');
+      answerClassificationResult.outcome.labelledAsCorrect = true;
+
+      spyOn(answerClassificationService, 'getMatchingClassificationResult')
+        .and.returnValue(answerClassificationResult);
+      spyOn(expressionInterpolationService, 'processHtml')
+        .and.callFake((html, envs) => html);
+      spyOn(focusManagerService, 'generateFocusLabel')
+        .and.returnValue('focusLabel');
+
+      questionPlayerEngineService.init(
+        multipleQuestionsBackendDict, successHandler, failHandler);
+
+      let createNewCardSpy = spyOn(
+        stateCardObjectFactory, 'createNewCard').and.returnValue(sampleCard);
+
+      questionPlayerEngineService.submitAnswer(
+        answer, interactionRulesService, successCallback);
+
+      expect(createNewCardSpy).toHaveBeenCalled();
+    });
+
     it('should not create next question if the existing ' +
```

Comment:
Please be careful with the terminology. You're not creating a question here.

Reply:
question --> card